### ISSUE TITLE
Prevent bad period creation after reporting period start date change

### ISSUE THREAD

**Issue by sanjuroj** [mercycorps/TolaActivity#703]
During the last SEA training, Myanmar wound up with an indicator with annual time periods with start dates a month apart.  The assumption is that there were two users and one created periodic targets at the same time as the reporting period was changed, which triggered a race condition, allowing for periodic targets with the old start date to be created at the same time the start date was changed.  Additional periodic targets were subsequently created with the new start date.  

To avoid this, there should be a backend check to prevent the creation of two sets of targets like this.  The current check is only on the front end.  This kind of error is tricky to identify after-the-fact so a better solution is to simply prevent it.

**Comment by jennyhavah** [comment]
@sanjuroj I remember a lot of discussion in Slack... What did we end up deciding to do here?

**Comment by sanjuroj** [comment]
The way I remember it, there was agreement that the race condition should be handled in some way.  There was less agreement on whether the additional step outlined in the third paragraph of this ticket was necessary.  I write up another ticket for closing the race condition and make this one focused on checking for indicator and program date mismatches.  We can argue about whether/how to do this one on slack or during a backlog meeting.

**Comment by jennyhavah** [comment]
@sanjuroj Ok to move this ticket to Poblano and we can continue arguing over there? ;-)

(Hoping to close Huckleberry.)

**Comment by sanjuroj** [comment]
@jennyhavah Yep, works for me.  BTW I already moved the remaining Huckleberry tickets to GP and closed Huckleberry.
